README.md

```diff
@@ -10,6 +10,133 @@
+
+result = independence_test(X, Y, method=Methods.FISHERZ)
+print("p-value:", result.pvalue, "Test statistic:", result.statistic)
+```
+
+Or for the kernel based independence test:
+
+```
+from pywhy_stats import Methods
+
+result = independence_test(X, Y, method=Methods.KCI)
+print("p-value:", result.pvalue, "Test statistic:", result.statistic)
+```
+
+For more information about the available methods, hyperparameters and other details, see the
+[documentation](https://py-why.github.io/pywhy-stats/stable/index.html).
+
+### Conditional independence test
+
+Similar to the unconditional independence test, we can use the same API to condition on another variable or set of 
+variables. First, let's generate a third variable $Z$ to condition on:
+
+```
+import numpy as np
+  
+rng = np.random.default_rng(0)
+Z = rng.standard_normal((200, 1))
+X = Z + rng.standard_normal(size=(200, 1))
+Y = np.exp(Z + rng.standard_normal(size=(200, 1)))
+```
+
+Here, $X$ and $Y$ are dependent due to $Z$. Running an unconditional independence test yields:
+
+```
+from pywhy_stats import independence_test
+ 
+result = independence_test(X, Y)
+print("p-value:", result.pvalue, "Test statistic:", result.statistic)
+```
+
+Again, the p-value is very small, indicating a high likelihood that $X$ and $Y$ are dependent. Now,
+let's condition on $Z$, which should render the variables as independent:
+
+```
+result = independence_test(X, Y, condition_on=Z)
+print("p-value:", result.pvalue, "Test statistic:", result.statistic)
+```
+
+We observe that the p-value isn't small anymore. Indeed, if the variables were independent, we would expect the p-value 
+to be uniformly distributed on $[0, 1]$.
+
+### (Conditional) k-sample test
+
+In certain settings, you may be interested in testing the invariance between k (conditional) distributions. For example, say you have data collected over the same set of variables (X, Y) from humans ($P^1(X, Y)$) and bonobos ($P^2(X, Y)$). You can determine if the conditional distributions $P^1(Y | X) = P^2(Y | X)$ using conditional two-sample test.
+
+First, we can create some simulated data that arise from two distinct distributions. However, the data generating Y is invariant across these two settings once we condition on X.
+
+```
+import numpy as np
+  
+rng = np.random.default_rng(0)
+X1 = rng.standard_normal((200, 1))
+X2 = rng.uniform(low=0.0, high=1.0, size=(200, 1))
+
+Y1 = np.exp(X1 + rng.standard_normal(size=(200, 1)))
+Y2 = np.exp(X2 + rng.standard_normal(size=(200, 1)))
+
+groups = np.concatenate((np.zeros((200, 1)), np.ones((200, 1))))
+X = np.concatenate((X1, X2))
+Y = np.concatenate((Y1, Y2))
+```
+
+We test the hypothesis that $P^1(Y | X) = P^2(Y | X)$ now with the following code.
+
+```
+from pywhy_stats import conditional_ksample
+
+# test that P^1(Y | X) = P^2(Y | X)
+result = conditional_ksample.kcd.condind(X, Y, groups)
+
+print("p-value:", result.pvalue, "Test statistic:", result.statistic)
+```
+
 # Documentation
```

Comment:
I would move Documentation and Installation (Dependencies and User Installation) sections above the QuickStart, since those are short and high-value info for someone.